website/docs/concepts/suspense.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Suspense"
+description: "Suspense for data fetching"
+---
+
+Suspense is a way to suspend component rendering whilst waiting a task
+to complete and a fallback (placeholder) UI is shown in the meanwhile.
+
+It can be used to fetch data from server, wait for tasks to be completed
+by an agent, or perform other background asynchronous task.
+
+Before suspense, data fetching usually happens after (Fetch-on-render) or before
+component rendering (Fetch-then-render).
+
+### Render-as-You-Fetch
+
+Suspense enables a new approach that allows components to initiate data request
+during the rendering process. When a component initiates a data request,
+the rendering process will become suspended and a fallback UI will be
+shown until the request is completed.
+
+The recommended way to use suspense is with hooks.
+
+```rust ,ignore
+use yew::prelude::*;
+
+#[function_component(Content)]
+fn content() -> HtmlResult {
+    let user = use_user()?;
+
+    Ok(html! {<div>{"Hello, "}{&user.name}</div>})
+}
+
+#[function_component(App)]
+fn app() -> Html {
+    let fallback = html! {<div>{"Loading..."}</div>};
+
+    html! {
+        <Suspense {fallback}>
+            <Content />
+        </Suspense>
+    }
+}
+```
+
+In the above example, the `use_user` hook will suspend the component
+rendering while user information is loading and a `Loading...` placeholder will
+be shown until `user` is loaded.
+
+To define a hook that suspends a component rendering, it needs to return
+a `SuspensionResult<T>`. When the component needs to be suspended, the
+hook should return a `Err(Suspension)` and users should unwrap it with
+`?` in which it will be converted into `Html`.
+
+```rust ,ignore
```

Comment:
Can we not ignore this?

Reply:
Not ignoring it would require including a complete hook implementation on every snippet.

I have included a complete example below.

Reply:
I wonder how long will it take for docusaurus to finally merge this in https://github.com/facebook/docusaurus/pull/5783
Then we could include all needed code and just hide it.